apps/backend/src/modules/user-follows/user-follows.controller.ts

```diff
@@ -0,0 +1,152 @@
+import { APIPath } from "~/libs/enums/enums.js";
+import { BaseController } from "~/libs/modules/controller/base-controller.module.js";
+import {
+	type APIHandlerOptions,
+	type APIHandlerResponse,
+} from "~/libs/modules/controller/libs/types/types.js";
+import { HTTPCode } from "~/libs/modules/http/http.js";
+import { type Logger } from "~/libs/modules/logger/libs/types/logger.type.js";
+import { UserFollowsApiPath } from "~/modules/user-follows/libs/enums/enums.js";
+import { type UserFollowsService } from "~/modules/user-follows/user-follows.service.js";
+
+import {
+	type UserFollowsFollowParametersDto,
+	type UserFollowsFollowRequestDto,
+	type UserFollowsUnfollowParametersDto,
+	type UserGetByIdItemResponseDto,
+} from "./libs/types/types.js";
+
+class UserFollowsController extends BaseController {
+	private userFollowsService: UserFollowsService;
+
+	public constructor(logger: Logger, userFollowsService: UserFollowsService) {
+		super(logger, APIPath.USERS);
+		this.userFollowsService = userFollowsService;
+
+		this.addRoute({
+			handler: this.follow.bind(this),
+			method: "POST",
+			path: UserFollowsApiPath.FOLLOW,
+		});
+
+		this.addRoute({
+			handler: this.unfollow.bind(this),
+			method: "DELETE",
+			path: UserFollowsApiPath.UNFOLLOW,
+		});
+	}
+
+	/**
+	 * @swagger
+	 * /users/{userId}/follow:
+	 *   post:
+	 *     tags:
+	 *       - Users
+	 *     summary: Follow a user
+	 *     parameters:
+	 *       - in: path
+	 *         name: userId
+	 *         required: true
+	 *         schema:
+	 *           type: integer
+	 *         description: ID of the user who is following
+	 *     requestBody:
+	 *       required: true
+	 *       content:
+	 *         application/json:
+	 *           schema:
+	 *             type: object
+	 *             properties:
+	 *               followingId:
+	 *                 type: integer
+	 *                 description: ID of the user to follow
+	 *                 example: 2
+	 *     responses:
+	 *       200:
+	 *         description: Successfully followed the user
+	 *       401:
+	 *         description: Unauthorized – userId does not match authorized user
+	 */
+
+	public async follow(
+		request: APIHandlerOptions<
+			{
+				body: UserFollowsFollowRequestDto;
+				params: UserFollowsFollowParametersDto;
+			} & { user: UserGetByIdItemResponseDto }
+		>,
+	): Promise<APIHandlerResponse> {
+		const followingId = Number(request.body.followingId);
+		const followerId = Number(request.params.userId);
+		const userId = Number(request.user?.id);
```

Comment:
We already know we will have user here, so let's cast it. As for the interface, we don't need to add user there, it's already included in type  and the same for unfollow
```suggestion
	public async follow(
		options: APIHandlerOptions<
			{
				body: UserFollowsFollowRequestDto;
				params: UserFollowsFollowParamsDto
			}
		>,
	): Promise<APIHandlerResponse> {
	    const { body, params, user } = options;
	    
		const followingId = Number(body.followingId);
		const followerId = Number(params.userId);
		const userId = (user as UserAuthResponseDto).id
```